schemas/swagger.yaml

```diff
@@ -54,31 +56,32 @@ paths:
       responses:
         200:
           description: Component revision curation.
+  /curations:
     patch:
-      summary: Patches the curation of a component revision.
+      summary: Patches the curation of a (set of) component revision.
       tags:
         - curations
-      parameters:
-        - $ref: '#/components/parameters/type'
-        - $ref: '#/components/parameters/provider'
-        - $ref: '#/components/parameters/namespace'
-        - $ref: '#/components/parameters/name'
-        - $ref: '#/components/parameters/revision'
+      requestBody:
+        required: true
+        content:
+          application/json:
+            schema:
+              $ref: /schemas/curations-1.0.json
```

Comment:
@jeffmcaffer @moranthomas I want to spend a few words around the issues we are facing around JSON schemas, starting from Jeff's comments: 

> This PR addresses all of these and gets the structure in the right format. There are still issues with the actual swagger being out of date and likely some incompatibilities with JSON schema and swagger (e.g., pattern properties). Those can be worked out in a subsequent PR

We have actually came up to the same conclusion.  
https://swagger.io/docs/specification/data-models/keywords/  Outlines the supported JSON Schema Keywords, and the Unsupported ones.
If we check for example https://github.com/clearlydefined/service/blob/master/schemas/curations-1.0.json, we'll notice that it makes use of the following **unsopported** keywords:
 - $schema
 - patternProperties

@jeffmcaffer  @moranthomas  I would recommend that We identify all the unsupported  properties and we then initiate a discussion to decide how to address this matter (e.g. creating the corresponding OpeAPI Schema, starting from something like https://github.com/wework/json-schema-to-openapi-schema, etc.)


@moranthomas can you ensure to spend some time doing it and open a new descriptive issue to address this matter?

Reply:
Agreed that this is not a complete solution. It does cover 90% of the cases:

* $schema is not that important afaick
* patternProperties is used in one place, curations

I'm fine heading in the direction you mentioned (see also the related article https://philsturgeon.uk/api/2018/04/13/openapi-and-json-schema-divergence-solved/). However, given we are 90% there, we could also wait for them to complete the OpenAPI 3.1 work. In the meantime we can put an example in the Swagger for curations and call it a day.

Short version: if it is a day of work to get the conversion approach to work, let's do it. If it's more, we likely have bigger fish to fry.